### ISSUE TITLE
virtualbox: registry: driver not found (and delete doesn't work)

### ISSUE THREAD

**Issue by tstromberg** [kubernetes/minikube#5295]
From #minikube Slack:

```
$ minikube start
😄  minikube v1.3.1 on Darwin 10.14.6
E0908 18:55:41.195263   51385 start.go:723] StartHost: Error loading existing host. Please try running [minikube delete], then run [minikube start] again.: registry: driver not found
```

```
$ minikube delete

💣  Failed to delete cluster: registry: driver not found

😿  Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
👉  https://github.com/kubernetes/minikube/issues/new/choose
```

My first thought was that this was xhyve, but it's virtualbox:

```$ minikube start --alsologtostderr -v=8
W0908 19:28:43.892842   52493 root.go:254] Error reading config file at /Users/jonas/.minikube/config/config.json: open /Users/jonas/.minikube/config/config.json: no such file or directory
I0908 19:28:43.894379   52493 notify.go:124] Checking for updates...
I0908 19:28:44.206587   52493 start.go:224] hostinfo: {"hostname":"Jonass-MacBook.local","uptime":1331455,"bootTime":1566632269,"procs":374,"os":"darwin","platform":"darwin","platformFamily":"","platformVersion":"10.14.6","kernelVersion":"18.7.0","virtualizationSystem":"","virtualizationRole":"","hostid":"982f17b3-0252-37fb-9869-88b3b1c77335"}
W0908 19:28:44.206744   52493 start.go:232] gopshost.Virtualization returned error: not implemented yet
😄  minikube v1.3.1 on Darwin 10.14.6
I0908 19:28:44.207409   52493 downloader.go:59] Not caching ISO, using https://storage.googleapis.com/minikube/iso/minikube-v1.3.0.iso
I0908 19:28:44.207544   52493 start.go:922] Saving config:
{
    "MachineConfig": {
        "KeepContext": false,
        "MinikubeISO": "https://storage.googleapis.com/minikube/iso/minikube-v1.3.0.iso",
        "Memory": 2000,
        "CPUs": 2,
        "DiskSize": 20000,
        "VMDriver": "virtualbox",
        "ContainerRuntime": "docker",
        "HyperkitVpnKitSock": "",
        "HyperkitVSockPorts": [],
        "DockerEnv": null,
        "InsecureRegistry": null,
        "RegistryMirror": null,
        "HostOnlyCIDR": "192.168.99.1/24",
        "HypervVirtualSwitch": "",
        "KVMNetwork": "default",
        "KVMQemuURI": "qemu:///system",
        "KVMGPU": false,
        "KVMHidden": false,
        "DockerOpt": null,
        "DisableDriverMounts": false,
        "NFSShare": [],
        "NFSSharesRoot": "/nfsshares",
        "UUID": "",
        "NoVTXCheck": false,
        "DNSProxy": false,
        "HostDNSResolver": true
    },
    "KubernetesConfig": {
        "KubernetesVersion": "v1.15.2",
        "NodeIP": "",
        "NodePort": 8443,
        "NodeName": "minikube",
        "APIServerName": "minikubeCA",
        "APIServerNames": null,
        "APIServerIPs": null,
        "DNSDomain": "cluster.local",
        "ContainerRuntime": "docker",
        "CRISocket": "",
        "NetworkPlugin": "",
        "FeatureGates": "",
        "ServiceCIDR": "10.96.0.0/12",
        "ImageRepository": "",
        "ExtraOptions": null,
        "ShouldLoadCachedImages": true,
        "EnableDefaultCNI": false
    }
}
I0908 19:28:44.207891   52493 cache_images.go:286] Attempting to cache image: gcr.io/k8s-minikube/storage-provisioner:v1.8.1 at /Users/jonas/.minikube/cache/images/gcr.io/k8s-minikube/storage-provisioner_v1.8.1
I0908 19:28:44.207969   52493 cache_images.go:286] Attempting to cache image: k8s.gcr.io/kube-apiserver:v1.15.2 at /Users/jonas/.minikube/cache/images/k8s.gcr.io/kube-apiserver_v1.15.2
I0908 19:28:44.208162   52493 cache_images.go:286] Attempting to cache image: k8s.gcr.io/pause:3.1 at /Users/jonas/.minikube/cache/images/k8s.gcr.io/pause_3.1
I0908 19:28:44.208240   52493 cache_images.go:286] Attempting to cache image: k8s.gcr.io/kube-proxy:v1.15.2 at /Users/jonas/.minikube/cache/images/k8s.gcr.io/kube-proxy_v1.15.2
I0908 19:28:44.208315   52493 cache_images.go:286] Attempting to cache image: k8s.gcr.io/k8s-dns-kube-dns-amd64:1.14.13 at /Users/jonas/.minikube/cache/images/k8s.gcr.io/k8s-dns-kube-dns-amd64_1.14.13
I0908 19:28:44.208216   52493 cache_images.go:286] Attempting to cache image: k8s.gcr.io/kube-scheduler:v1.15.2 at /Users/jonas/.minikube/cache/images/k8s.gcr.io/kube-scheduler_v1.15.2
I0908 19:28:44.208467   52493 cache_images.go:286] Attempting to cache image: k8s.gcr.io/kubernetes-dashboard-amd64:v1.10.1 at /Users/jonas/.minikube/cache/images/k8s.gcr.io/kubernetes-dashboard-amd64_v1.10.1
I0908 19:28:44.208246   52493 cache_images.go:286] Attempting to cache image: k8s.gcr.io/kube-controller-manager:v1.15.2 at /Users/jonas/.minikube/cache/images/k8s.gcr.io/kube-controller-manager_v1.15.2
I0908 19:28:44.208432   52493 cache_images.go:286] Attempting to cache image: k8s.gcr.io/k8s-dns-dnsmasq-nanny-amd64:1.14.13 at /Users/jonas/.minikube/cache/images/k8s.gcr.io/k8s-dns-dnsmasq-nanny-amd64_1.14.13
I0908 19:28:44.208439   52493 cache_images.go:286] Attempting to cache image: k8s.gcr.io/coredns:1.3.1 at /Users/jonas/.minikube/cache/images/k8s.gcr.io/coredns_1.3.1
I0908 19:28:44.208451   52493 cache_images.go:286] Attempting to cache image: k8s.gcr.io/k8s-dns-sidecar-amd64:1.14.13 at /Users/jonas/.minikube/cache/images/k8s.gcr.io/k8s-dns-sidecar-amd64_1.14.13
I0908 19:28:44.208384   52493 cache_images.go:286] Attempting to cache image: k8s.gcr.io/etcd:3.3.10 at /Users/jonas/.minikube/cache/images/k8s.gcr.io/etcd_3.3.10
I0908 19:28:44.208671   52493 cache_images.go:286] Attempting to cache image: k8s.gcr.io/kube-addon-manager:v9.0 at /Users/jonas/.minikube/cache/images/k8s.gcr.io/kube-addon-manager_v9.0
I0908 19:28:44.209502   52493 cluster.go:98] Skipping create...Using existing machine configuration
E0908 19:28:44.211844   52493 start.go:723] StartHost: Error loading existing host. Please try running [minikube delete], then run [minikube start] again.: registry: driver not found
I0908 19:28:44.212446   52493 utils.go:127] non-retriable error: Error loading existing host. Please try running [minikube delete], then run [minikube start] again.: registry: driver not found
W0908 19:28:44.212585   52493 exit.go:99] Unable to start VM: Error loading existing host. Please try running [minikube delete], then run [minikube start] again.: registry: driver not found
💣  Unable to start VM: Error loading existing host. Please try running [minikube delete], then run [minikube start] again.: registry: driver not found
😿  Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
👉  https://github.com/kubernetes/minikube/issues/new/choose
```

I compared the configuration to my own and it seemed proper.

The workaround was:

`rm -Rf  ~/.minikube/profiles/minikube ~/.minikube/machines/minikube`

We should make sure `minikube delete` still works in this case.

**Comment by afbjorklund** [comment]
I'm not sure if it was "virtualbox", the log says `Skipping create...Using existing machine configuration`

Unfortunately the old machine configuration seems to have been deleted, per commands above.

**Comment by afbjorklund** [comment]
The important configuration is the old one in `machines`, not the new one in `profiles`.

**Comment by tstromberg** [comment]
I believe this was resolved by minikube v1.4.
